packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -174,6 +174,125 @@ class Plugin {
     );
   }
 
+  /// Queries the `pubpsec` and returns the `platforms` sub map.
+  ///
+  /// Throws ToolExit if cannot find the `platforms:` map.
+  static YamlMap getPlatformsYamlMap(YamlMap pubspec) {
+    if (pubspec == null) {
```

Comment:
Assuming that someone is migrating an old plugin and trying to use add-platform, it would be nice to at least link to page explaining the formatting

Reply:
done. We only need to show  this error if the `platforms` map is missing or not valid, other failing scenarios are just general invalid pubspec.